streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -114,18 +115,28 @@ public StreamTask(TaskId id,
         // to corresponding source nodes in the processor topology
         partitionQueues = new HashMap<>();
 
-        TimestampExtractor timestampExtractor = config.getConfiguredInstance(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
+        final TimestampExtractor defaultTimestampExtractor = config.getConfiguredInstance(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
 
         for (TopicPartition partition : partitions) {
             SourceNode source = topology.source(partition.topic());
```

Comment:
I am still confused, about `source` being `null`. In the original code (L121) `source` is handed to `createRecrodQueue` and must not be `null` -- because this was never an issues before, I am still puzzled. why it is now.